### PR TITLE
ZENKO-3355 Update UX on replication GUI

### PR DESCRIPTION
**Pull request by theocerutti**
Hello!

> More vertical space between the header "Create New Rule" and "Rule Type" (24px)

There was already a 40px margin-bottom. Should I lower the margin to 24px?

>     Add the tooltips next to the label that needs explanation (+ Terry to provide the content)
waiting for Terry

### CODE REVIEW

**Review by Cuervino — CHANGES_REQUESTED** [review]
> 1.         Change the string to "Create New Rule" (with the correct Pascal Case)
OK

> 2.         More vertical space between the header "Create New Rule" and "Rule Type" (24px)
Yes please make it 24 px (looks better)

> 3.         Add "All * are mandatory fields" between this title and "Rule Type", and add a "*" next to all the required fields ("Rule name", Source"Bucket Name", Destination"BucketName")
Ok for the creation form, but the "*" should be present only on the creation stage, not on the edition. So please remove the "*" from the "configuration" tab once the rule has been created.

> 4.         Add an icon next to the rule type for instance <i className="fas fa-coins" /> (see in core-ui https://scality.github.io/core-ui/?path=/story/style-icons--page)
The icon should be on the left (not on the right) - 8px of space between icon & label.

> 5.         "Rule Name" instead of "Name"
Ok

> 6.         States should be with the first letter in Uppercase (Active / Inactive), same in the table.
Ok

> 7.         Workflow type should be with the first letter in Uppercase in the table (here "Replication" instead of "replication")
Ok

> 8.         Add the tooltips next to the label that needs explanation (+ Terry to provide the content)
Terry to wait - if it's the only missing point, we could merge and add the tooltips in another PR.

> 9.         24px spacing between buttons "Cancel" and "Create"
Great

File: src/react/ui-elements/FormLayout.jsx [hunk 1]
```diff
@@ -32,10 +32,16 @@ export const Title = styled.div`
     display: flex;
 
     text-transform: capitalize;
-    margin-bottom: 40px;
+    margin-bottom: 24px;
     font-size: 19px;
 `;
```
Comment (theocerutti) [inline src/react/ui-elements/FormLayout.jsx]: Just to be sure: I changed the margin-bottom to 24px for ALL titles (<Title/> Component) is that what I needed to do or is it just in this case?

Comment (Cuervino) [reply src/react/ui-elements/FormLayout.jsx]: No, just for the case of Replication, which is a specific one.

**Review by Cuervino — APPROVED** [review]
Good!
The tooltips will be done in another issue/PR.

**Comment by theocerutti** [comment]
@bert-e approve

File: src/react/workflow/WorkflowContent.jsx [hunk 2]
```diff
@@ -23,6 +24,15 @@ type Props = {
 
 export const InfoWarning = ({ title }: { title: string}) => <Warning iconClass='fas fa-2x fa-info-circle' title={title} />;
 
+const ReplicationTitle = styled(Title)`
```
Comment (nicolas2bert) [inline src/react/workflow/WorkflowContent.jsx]: These styled components can be moved to the `../ui-elements//TableKeyValue2` and be renamed `Title` and `Subtitle`. That would help UX consistency.

File: src/react/workflow/WorkflowList.jsx [hunk 3]
```diff
@@ -8,8 +8,13 @@ import { ListSection } from '../ui-elements/ListLayout3';
 import React from 'react';
 import type { Rules } from '../../types/config';
 import { push } from 'connected-react-router';
+import styled from 'styled-components';
 import { useDispatch } from 'react-redux';
 
+const WorkflowTypeCell = styled.span`
```
Comment (nicolas2bert) [inline src/react/workflow/WorkflowList.jsx]: We might want to create a `ui-elements/Utility.js` file and import it from there for reusability.

File: src/react/workflow/replication/Replication.jsx [hunk 4]
```diff
@@ -28,6 +28,10 @@ const ReplicationContainer = styled.div`
     }
 `;
 
+const ReplicationIcon = styled.i`
```
Comment (nicolas2bert) [inline src/react/workflow/replication/Replication.jsx]: Let's avoid having too many styled components that will affect readability. Can we set an CSS rule for `<T.Key>`'s icon in `../../ui-elements/TableKeyValue2` ?

File: src/react/workflow/replication/Replication.jsx [hunk 5]
```diff
@@ -118,7 +125,7 @@ function Replication({ streams, bucketList, locations, workflowDetails, showEdit
                     </T.GroupName>
                     <T.GroupContent>
                         <T.Row>
-                            <T.Key> Name </T.Key>
+                            <T.Key required={createMode}> Rule Name </T.Key>
```
Comment (nicolas2bert) [inline src/react/workflow/replication/Replication.jsx]: Name is not required, we generate one for you if missing. CC @Cuervino

**Review by nicolas2bert — COMMENTED** [review]
It looks like your tests are not passing, you can fix them running:
- for linting: npm run test:lint
- for unit: npm run test

File: src/react/workflow/replication/Replication.jsx (hunk 5 above)
Comment (Cuervino) [reply src/react/workflow/replication/Replication.jsx]: Ah, well spotted.

File: src/react/workflow/WorkflowContent.jsx (hunk 2 above)
Comment (theocerutti) [reply src/react/workflow/WorkflowContent.jsx]: done

File: src/react/workflow/WorkflowList.jsx (hunk 3 above)
Comment (theocerutti) [reply src/react/workflow/WorkflowList.jsx]: done

File: src/react/workflow/replication/Replication.jsx (hunk 4 above)
Comment (theocerutti) [reply src/react/workflow/replication/Replication.jsx]: done

File: src/react/workflow/replication/Replication.jsx (hunk 5 above)
Comment (theocerutti) [reply src/react/workflow/replication/Replication.jsx]: done

**Review by Cuervino — APPROVED** [review]
LGTM

File: src/react/ui-elements/FormLayout.jsx [hunk 6]
```diff
@@ -36,9 +36,9 @@ import styled from 'styled-components';
 export const Title = styled.div`
     display: flex;
     text-transform: capitalize;
-    color: ${props => props.theme.brand.textPrimary};
     margin-bottom: 40px;
-    font-size: 24px;
+    font-size: 19px;
```
Comment (nicolas2bert) [inline src/react/ui-elements/FormLayout.jsx]: should we use the theme's sizes?

Comment (Cuervino) [reply src/react/ui-elements/FormLayout.jsx]: yes, it's always better.

**Review by nicolas2bert — APPROVED** [review]

**Pull request merged**